datafusion/src/physical_plan/json.rs

```diff
@@ -202,16 +202,11 @@ impl NdJsonExec {
         max_records: Option<usize>,
     ) -> Result<Schema> {
         let mut schemas = Vec::new();
-        let mut records_to_read = max_records.unwrap_or(usize::MAX);
-        while records_to_read > 0 && !filenames.is_empty() {
+        let lines_per_file = max_records.map(|x| x / filenames.len());
```

Comment:
One edge case I think this doesn't take care of:

* Integer division by `filenames.len()` is 0 when the number of files is bigger than the number of records.

Reply:
Any idea what we should do in that case? take one line per file?

Reply:
I think we should round up and either read a bit more lines than specified or check whether the limit has been reached.

Reply:
Round up to 1 seems like the only choice if we want to spread out the read. If we don't check whether the limit has been reached, this would change the semantic of the argument in both ways, i.e. it might read **more** or **less** than specified value. So the prefix `max_` would not be accurate anymore. If we want to go that route, it would be good to rename the argument or implement this behavior in a new method.